doc/helpers/responsive-extension.md

```diff
@@ -0,0 +1,109 @@
+| ---------- | ---------------- | ---------------------------------------------------------- |
+| Narrowest  | object           | Value to be used when the screen size is at its narrowest. |
+| Narrow     | object           | Value to be used when the screen size is narrow.           |
+| Normal     | object           | Value to be used when the screen size is normal.           |
+| Wide       | object           | Value to be used when the screen size is wide.             |
+| Widest     | object           | Value to be used when the screen size is at its widest.    |
+| Layout     | ResponsiveLayout | Overrides the screen size thresholds/breakpoints.          |
+
+### Usage
+```xml
+xmlns:utu="using:Uno.Toolkit.UI"
+...
+<Page.Resources>
+	<SolidColorBrush x:Key="NarrowRed">Crimson</SolidColorBrush>
+	<SolidColorBrush x:Key="WideBlue">Blue</SolidColorBrush>
+</Page.Resources>
+...
+<TextBlock Text="Asd"
+		   Background="{utu:Responsive Narrow={StaticResource NarrowRed}, Wide={StaticResource WideBlue}}" />
+```
+
+## Remarks
+**Property type limitation**: Content property values other than strings must be appropriately typed for the markup extension to interpret them correctly.
+In the usage example above, the values `NarrowRed` and `WideBlue` are properly typed as they refer to the `StaticResource` keys defined in `Page.Resources`.
+For instance, using `Background={utu:Responsive Narrow=SkyBlue, Wide=Pink}` would be incorrect, while the string literal usage under `<TextBlock Text="{utu:Responsive Narrow='Narrow format', Wide='Wide format'}" />` is accepted.
+
+**Platform limitation**: The ability to update property values when the window size changes is only available on targets other than Windows UWP.
+Due to a limitation of the UWP API (Windows target only), the `MarkupExtension.ProvideValue(IXamlServiceProvider)` overload is unavailable, which is required to continuously update the value.
+Because of this, the markup extension will only provide the initial value, and will not respond to window size changes.
+
+**Initialization**: The `ResponsiveHelper` needs to be hooked up to the window's `SizeChanged` event in order for it to receive updates when the window size changes.
+This is typically done in the `App.xaml.cs` file, where you can get the current window and call the `HookupEvent` method on the `ResponsiveHelper`.
+It is important to do this when the app launches, otherwise the `ResponsiveExtension` won't be able to update the property values when the window size changes.
+
+Here is an example of how this might be achieved:
+
+```cs
+protected override void OnLaunched(LaunchActivatedEventArgs args)
+{
+#if NET6_0_OR_GREATER && WINDOWS && !HAS_UNO
+	MainWindow = new Window();
+#else
+	MainWindow = Microsoft.UI.Xaml.Window.Current;
+#endif
+	// ...
+	var helper = Uno.Toolkit.UI.Helpers.ResponsiveHelper.GetForCurrentView();
+	helper.HookupEvent(MainWindow);
```

Comment:
Can't this be done for us in the Toolkit init process? 

There's no way around this?

@Xiaoy312

Reply:
if you can access the window somehow, yes of course
otherwise, no

Reply:
@MartinZikmund are we gonna have some sort of way of accessing the current Window from the scope of a given UIElement?